examples/pipeline/run_text-generation.py

```diff
@@ -0,0 +1,54 @@
+import argparse
+import time
+
+import torch
+from transformers import AutoConfig, AutoModelForCausalLM, AutoTokenizer, pipeline
+
+from optimum.intel import inference_mode
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--model_id", default=None, type=str)
+args = parser.parse_args()
+
+kwargs = dict()
+kwargs["torch_dtype"] = torch.bfloat16
+kwargs["use_cache"] = True
+kwargs["low_cpu_mem_usage"] = True
+kwargs["return_dict"] = False
+model_id = args.model_id
+model = AutoModelForCausalLM.from_pretrained(model_id, **kwargs)
+tokenizer = AutoTokenizer.from_pretrained(model_id)
+model = model.eval().to(torch.device("cpu"))
+model = model.to(memory_format=torch.channels_last)
+
+# 32 tokens
+input_seq = "DeepSpeed is a machine learning framework for deep neural networks and deep reinforcement learning. It is written in C++ and is available for Linux, Mac OS X,"
+
+print("Input sequence is: ")
+print(input_seq)
+
+generate_kwargs = dict(max_new_tokens=32, do_sample=False, num_beams=4, num_beam_groups=1)
```

Comment:
I don't think an example would be relevant here, but would rather have it in the documentation instead what do you think ? Also it would make sense to add a test as well !